EXPLORATORY DATA ANALYSIS

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("../data/suppliers.csv")
print(df.shape)
print(df.dtypes)
print(df.isnull().sum())

(1200, 39)
supplier_country                          str
political_stability_index             float64
infrastructure_score                  float64
natural_disaster_frequency            float64
corruption_index                      float64
conflict_score                        float64
sanctions_exposure                       bool
partial_export_restriction               bool
diplomatic_relations_score            float64
distance_km                             int64
is_landlocked                            bool
supplier_tenure_years                   int64
contract_type                             str
communication_responsiveness_hours    float64
num_alternative_suppliers               int64
supplier_tier                             str
quality_incidents                       int64
audit_score                           float64
iso_certified                            bool
regulatory_violations                   int64
financial_stability_index             float64
debt_to_equity         

In [ ]:
df["contract_type"]  = df["contract_type"].astype("category")
df["supplier_tier"]  = pd.Categorical(df["supplier_tier"], categories=["Tier 1", "Tier 2", "Tier 3"], ordered=True)
df["transport_mode"] = df["transport_mode"].astype("category")

#CSV doesn't preserve categorical dtype — it saves everything as strings, 
# so to re-casting after loading is needed, that's why it is not in the first notebook

In [3]:
labels = {0: "None", 1: "Low", 2: "Medium", 3: "High", 4: "Critical"}
df["risk_label"] = df["risk_level"].map(labels)

counts = df["risk_level"].value_counts().sort_index()
for k, v in counts.items():
    print(k, labels[k], v)

0 None 296
1 Low 131
2 Medium 218
3 High 171
4 Critical 384


1200 suppliers across 5 risk levels, with Critical (384) most common and Low (131) least common. Class imbalance will be addressed with class_weight="balanced" in all models during training.

In [4]:
medium_archetype = df[df["archetype"] == 2]
print(medium_archetype["risk_level"].value_counts().sort_index())

risk_level
1     61
2    210
3     66
4     63
Name: count, dtype: int64


Out of 400 Medium archetype suppliers, 210 were correctly classified as Medium. 61 leaked to Low, 66 to High, 63 to Critical.

In [5]:
for i in range(5):
    print(f"Archetype {i}:", df[df["archetype"] == i]["risk_level"].value_counts().sort_index().to_dict())

Archetype 0: {0: 187, 1: 1, 3: 10, 4: 2}
Archetype 1: {0: 109, 1: 69, 3: 9, 4: 13}
Archetype 2: {1: 61, 2: 210, 3: 66, 4: 63}
Archetype 3: {2: 8, 3: 83, 4: 109}
Archetype 4: {3: 3, 4: 197}
